source/fundamentals/builders/updates.txt

```diff
@@ -11,3 +11,505 @@ Updates Builders
+   :language: java
+   :dedent:
+   :start-after: begin mulUpdate
+   :end-before: end mulUpdate
+
+The example above updates the original document to the following state:
+   
+.. code-block:: json
+   :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      "qty": 10, 
+      "vendor": [ "A", "D", "M" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021 
+   }
+
+Rename
+~~~~~~
+Use the :java-core-api:`rename() <com/mongodb/client/model/Updates.html#rename(java.lang.String,java.lang.String)>`
+method to rename the value of a field in an update operation.
+
+The following example renames the ``qty`` field to "quantity":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Updates.java
+   :language: java
+   :dedent:
+   :start-after: begin renameUpdate
+   :end-before: end renameUpdate
+   
+The example above updates the document to the following state:
+   
+.. code-block:: json
+   :copyable: false
+      
+   {
+      "_id": 1,
+      "color": "red", "
+      vendor": [ "A", "D", "M" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021, 
+      "quantity": 5
+   }
+
+Min
+~~~
+Use the :java-core-api:`min() <com/mongodb/client/model/Updates.html#min(java.lang.String,TItem)>`
+method to update the value of a field with the smaller number of the two
+specified in an update operation.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Updates.java
+   :language: java
+   :dedent:
+   :start-after: begin minUpdate
+   :end-before: end minUpdate
+
+The example above updates the document to the following state:
+   
+.. code-block:: json
+   :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      qty": 2, 
+      "vendor": [ "A", "D", "M" ], 
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021
+   }
+
+Max
+~~~
+Use the :java-core-api:`max() <com/mongodb/client/model/Updates.html#max(java.lang.String,TItem)>`
+method to update the value of a field with the larger number of the two
+specified in an update operation.
+
+The following example sets the value of the ``qty`` field to the
+maximum of its current value and "8": 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Updates.java
+   :language: java
+   :dedent:
+   :start-after: begin maxUpdate
+   :end-before: end maxUpdate
+
+The example above updates the document to the following state:
+   
+.. code-block:: json
+  :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      "qty": 8, 
+      "vendor": [ "A", "D", "M" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021
+   }
+
+Current Date
+~~~~~~~~~~~~
+Use the :java-core-api:`currentDate() <com/mongodb/client/model/Updates.html#currentDate(java.lang.String)>`
+method to assign the value of a field in an update operation to the
+current date as a :manual:`date </reference/bson-types/#document-bson-type-date>`
+or :manual:`timestamp <reference/bson-types/#document-bson-type-timestamp>`.
```

Comment:
To update it to a timestamp, you have to use `currentTimestamp` I think

Reply:
https://docs.mongodb.com/manual/reference/operator/update/currentDate/

Reply:
Right, but the currentDate builder doesn't work that way: it only generates dates.  You have to use currentTimestamp to generate a timestamp.  Maybe not worth mentioning that though since you rarely need a timestamp.

Reply:
What did you end up doing with this one?

Reply:
I left it as it is since this is what https://mongodb.github.io/mongo-java-driver/4.2/builders/updates/ has for currentDate.  Would you like it to be changed to currentTimestamp?

Reply:
Yeah, I think current docs are wrong.  currentDate helper gives you a date, currentTimestamp gives you a timestamp.

Reply:
I just tested both methods.
currentTimestamp returns: lastModified=Timestamp{value=6942526372958961671, seconds=1616432884, inc=7}
currentDate returns: lastModified=Mon Mar 22 13:08:04 EDT 2021

I think currentDate performs as expected.  So to clarify, you want currentTimestamp to replace the currentDate section?

Reply:
How about two sections, one for each helper method?